bedevere/backport.py

```diff
@@ -41,83 +49,63 @@ async def _remove_backport_label(gh, original_issue, branch, backport_pr_number)
-async def manage_labels(event, gh, *args, **kwargs):
-    if event.data["action"] == "edited" and "title" not in event.data["changes"]:
+async def manage_labels(gh, *args, **kwargs):
+    with open(os.environ["GITHUB_EVENT_PATH"]) as f:
+        event = json.load(f)
+    if event.get("action") == "edited" and "title" not in event.get("changes"):
         return
-    pull_request = event.data["pull_request"]
+    pull_request = event["pull_request"]
     title = util.normalize_title(pull_request['title'],
-                                 pull_request['body'])
+                                pull_request['body'])
     title_match = TITLE_RE.match(title)
     if title_match is None:
         return
     branch = title_match.group('branch')
     original_pr_number = title_match.group('pr')
-
-    original_issue = await gh.getitem(event.data['repository']['issues_url'],
-                                      {'number': original_pr_number})
+    
+    original_issue = await gh.getitem(event['repository']['issues_url'],
+                                    {'number': original_pr_number})
     await _remove_backport_label(gh, original_issue, branch,
-                                 event.data["number"])
+                                event["number"])
 
-    backport_issue = await util.issue_for_PR(gh, pull_request)
+    backport_issue = await issue_for_PR(gh, pull_request)
     await _copy_over_labels(gh, original_issue, backport_issue)
 
-
-def is_maintenance_branch(ref):
-    """
-    Return True if the ref refers to a maintenance branch.
-
-    >>> is_maintenance_branch("3.11")
-    True
-    >>> is_maintenance_branch("main")
-    False
-    >>> is_maintenance_branch("gh-1234/something-completely-different")
-    False
-    """
-    maintenance_branch_pattern = r'\d+\.\d+'
-    return bool(re.fullmatch(maintenance_branch_pattern, ref))
-
-
-@router.register("pull_request", action="opened")
-@router.register("pull_request", action="reopened")
-@router.register("pull_request", action="edited")
-@router.register("pull_request", action="synchronize")
-async def validate_maintenance_branch_pr(event, gh, *args, **kwargs):
+async def validate_maintenance_branch_pr(gh, *args, **kwargs):
     """Check the PR title for maintenance branch pull requests.
 
     If the PR was made against maintenance branch, and the title does not
     match the maintenance branch PR pattern, then post a failure status.
 
     The maintenance branch PR has to start with `[X.Y]`
     """
-    if event.data["action"] == "edited" and "title" not in event.data["changes"]:
+    with open(os.environ["GITHUB_EVENT_PATH"]) as f:
+        event = json.load(f)
+    if event.get("action") == "edited" and "title" not in event.get("changes"):
         return
-    pull_request = event.data["pull_request"]
+    pull_request = event["pull_request"]
     base_branch = pull_request["base"]["ref"]
 
-    if not is_maintenance_branch(base_branch):
+    if base_branch == "main":
```

Comment:
This should also use `not is_maintanance_branch`.
You should double check that there aren't other updates that got revert during the merge (or possibly re-do the merge).